poplar.mk

```diff
@@ -0,0 +1,286 @@
+################################################################################
+# Following variables defines how the NS_USER (Non Secure User - Client
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+COMPILE_NS_USER   ?= 64
+override COMPILE_NS_KERNEL := 64
+COMPILE_S_USER    ?= 32
+COMPILE_S_KERNEL  ?= 64
+
+################################################################################
+# Includes
+################################################################################
+include common.mk
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+ifeq ($(DEBUG),1)
+ARM_TF_BUILD		?= debug
+else
+ARM_TF_BUILD		?= release
+endif
+
+OUT_PATH		?= $(ROOT)/out
+
+ARM_TF_PATH		?= $(ROOT)/arm-trusted-firmware
+LLOADER_PATH		?= $(ROOT)/l-loader
+UBOOT_PATH		?= $(ROOT)/u-boot
+OPTEE_PATH		?= $(ROOT)/optee_os
+OPTEE_CLIENT_PATH	?= $(ROOT)/optee_client
+OPTEE_CLIENT_EXPORT	?= $(OPTEE_CLIENT_PATH)/out/export
+OPTEE_TEST_PATH		?= $(ROOT)/optee_test
+LINUX_PATH		?= $(ROOT)/linux
+TOOLS_PATH		?= $(ROOT)/poplar-tools
+
+BL1_BIN			?= $(ARM_TF_PATH)/build/poplar/$(ARM_TF_BUILD)/bl1.bin
+FIP_BIN			?= $(ARM_TF_PATH)/build/poplar/$(ARM_TF_BUILD)/fip.bin
+LLOADER_BIN		?= $(LLOADER_PATH)/l-loader.bin
+
+LINUX_DTB		?= $(LINUX_PATH)/arch/arm64/boot/dts/hisilicon/hi3798cv200-poplar.dtb
+
+UBOOT_BIN		?= $(UBOOT_PATH)/u-boot.bin
+OPTEE_BIN		?= $(OPTEE_PATH)/out/arm/core/tee-header_v2.bin
+OPTEE_BIN_EXTRA1	?= $(OPTEE_PATH)/out/arm/core/tee-pager_v2.bin
+OPTEE_BIN_EXTRA2	?= $(OPTEE_PATH)/out/arm/core/tee-pageable_v2.bin
+
+ROOTFS_BIN		?= linaro-stretch-developer-*.tar.gz
+ROOTFS_URL		?= http://snapshots.linaro.org/debian/images/stretch/developer-arm64/latest/$(ROOTFS_BIN)
+
+PKG_OPTEE_VERSION	?= $(shell cd $(OPTEE_OS_PATH) && git describe)
```

Comment:
Need to end with `-0`. See https://github.com/OP-TEE/build/pull/154#discussion_r121323471 for details.

Other than that, `Reviewed-by: Victor Chong <victor.chong@linaro.org>`

Reply:
fixed, thanks!